spec/models/spree/order_spec.rb

```diff
@@ -283,6 +283,13 @@
       order.completed_at = Time.zone.now
       expect(order.can_cancel?).to be_truthy
     end
+
+    it "should cancel order if product is soft deleted" do
+      o = FactoryBot.create(:completed_order_with_totals)
+      o.line_items.first.variant.product.tap(&:destroy)
+      o.cancel!
+      expect(Spree::Order.by_state(:canceled)).to include o
+    end
```

Comment:
Nice! I've just got some syntax suggestions (let/create syntax and avoiding single-letter-variables):

```suggestion
    context "with a deleted product" do
      let(:order) { create(:completed_order_with_totals) }
      
      it "should cancel order without error" do
        order.line_items.first.variant.product.tap(&:destroy)
        order.cancel!
        expect(Spree::Order.by_state(:canceled)).to include order
      end
    end  
```

Reply:
great suggestion, thanks